pkg/templates/template_parser.go

```diff
@@ -11,7 +11,12 @@ type ITemplateParser interface {
 
 func (tp *TemplateParser) ParseTemplate(templateFileName []string, data interface{}) (string, error) {
 	var parsedTemplate string
-	t, err := template.ParseFiles(templateFileName...)
+
+	funcMap := template.FuncMap{
```

Comment:
you dont need to modify the `ParseTemplate` function.A function is defined on a type.
A type backs a template.

So whichever type you are binding to the template should have the function you need to call.